lib/cli/etcher.js

```diff
@@ -18,61 +18,115 @@
 
 const _ = require('lodash');
 const Bluebird = require('bluebird');
+const inquirer = require('inquirer');
 const isElevated = Bluebird.promisify(require('is-elevated'));
 const visuals = require('resin-cli-visuals');
-const form = require('resin-cli-form');
 const drivelist = Bluebird.promisifyAll(require('drivelist'));
+const chalk = require('chalk');
+const imageStream = require('etcher-image-stream');
 const writer = require('./writer');
 const utils = require('./utils');
 const options = require('./cli');
+const DynamicList = require('./dynamic-list');
+const DriveScanner = require('./drive-scanner');
 const EXIT_CODES = require('../src/exit-codes');
 
-isElevated().then((elevated) => {
+inquirer.registerPrompt('dynamicList', DynamicList);
+
+const scanner = new DriveScanner(2000);
+
+const driveToChoice = (drive, minSizeToEnable) => {
+  const size = drive.size / 1000000000;
+
+  let displayName = `${drive.device} (${size.toFixed(1)} GB) - ${drive.description}`;
+
+  if (drive.system) {
+    displayName += chalk.red(' FIXED DRIVE');
+  }
+
+  const choice = {
+    name: displayName,
+    value: drive.device
+  };
+  if (drive.size < minSizeToEnable) {
```

Comment:
Oh, I see, my bad.

> Anyway, by reusing logic from the SelectionStateModel you meant actually importing it here? Or maybe refactoring some useful methods to a common module?

It'd be really cool if we could re-use it as it is, however `SelectionStateModel` is an Angular service, and can't be used outside such framework. I propose extracting all the "static" functions that module contains (that don't interact with the application state), such as `isSystemDrive()`, `isDriveLocked()`, etc; as a separate NodeJS module we can also import in the CLI.

Some things to keep in mind:

- These functions contain tests, so the tests will have to be extracted as well.
- The extracted module will still need to be exposed as an Angular service so the Angular components that currently use these functions can still do so (you can collaborate with Shou on this one).

The current approach to re-usable code between the GUI and CLI is to put it on `lib/src` (I know the name is not ideal, we should probably rename to `lib/shared`), so the extracted functionality can live there, and then there can be another small module in `lib/gui` that simply requires that logic and re-exposes it as an Angular service.

In the future, we want to have a re-usable module that handles the "labels" we put to drives (such as system, locked, etc) and also be able to automatically compute a list of them (so we don't hardcode the label names anywhere), so that would be the perfect place to include that.

Let's make those changes as another PR, so its easier to review, and then you can rebase here.